### PR TITLE
Move Consumer Management to JetStream

### PR DESCRIPTION
**Pull request by Jarema**
First step to get rid of Consumer as it is now.

### CODE REVIEW

File: src/jetstream.rs [hunk 1]
```diff
@@ -780,7 +812,35 @@ impl JetStream {
...
 
-        Consumer::existing::<&str, ConsumerConfig>(self.clone(), stream, config)
+        self.existing::<&str, ConsumerConfig>(&stream, config)
+    }
+
+    /// Instantiate a `JetStream` `Consumer`. Performs a check to see if the consumer
+    /// already exists, and creates it if not. If you want to use an existing
+    /// `Consumer` without this check and creation, use the `existing`
+    /// method.
+    pub fn create_or_open<S, C>(&self, stream: S, cfg: C) -> io::Result<Consumer>
```
Comment (derekcollison) [inline src/jetstream.rs:822]: We use bind in other clients like the Go one.

Comment (Jarema) [reply src/jetstream.rs]: Renamed to `create_or_bind` to reflect that as `bind` does not try to create a consumer if it doesn't exist.

**Review by derekcollison — APPROVED** [review]
LGTM

File: tests/jetstream.rs [hunk 2]
```diff
@@ -187,8 +185,9 @@ fn jetstream_libdoc_test() {
     nc.publish("my_stream", "3").unwrap();
     nc.publish("my_stream", "4").unwrap();
 
-    let mut consumer =
-        Consumer::create_or_open(js, "my_stream", "existing_or_created_consumer").unwrap();
+    let mut consumer = js
+        .create_or_bind("my_stream", "existing_or_created_consumer")
```
Comment (caspervonb) [inline tests/jetstream.rs:189]: Naming is a bit on the generic side as neither create and bind are "known operations" within the module.

**Pull request merged**